awesomemotive/missed-scheduled-posts-publisher
# Setup PHPUnit

This PR setup PHPUnit for the plugin.

To keep the setup minimal, I use this prebuilt Docker image (https://github.com/humanmade/plugin-tester) which comes with WordPress and PHPUnit configured.

The PR also adds a [GitHub Action](https://github.com/awesomemotive/missed-scheduled-posts-publisher/blob/setup/php-unit/.github/workflows/php-test.yml) to run the tests in CI. The tests will run on CI on commits or PRs against the main branch. This has been tested on my fork here.

## How to test
- Pull the PR locally
- Run `composer install` (there seems to be errors with Composer v2. Run `composer self-update --1` to switch to Composer v1)
- Make sure you have Docker running
- Runs `docker run --rm -v "$PWD:/code" humanmade/plugin-tester`

The last commands should show that the [sample test](https://github.com/awesomemotive/missed-scheduled-posts-publisher/blob/setup/php-unit/tests/class-sample-test.php) is passing.
